src/components/admin/hotels/save-hotel/index.js

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import {
+    Button,
+    Box,
+    Container,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    FormControl,
+    Grid,
+    TextField,
+    Typography,
+} from '@material-ui/core';
+import { Rating } from '@material-ui/lab';
+import HotelIcon from '@material-ui/icons/Hotel';
+
+import ManageAttachments from './manage-attachments';
+import HotelClient from '../../../../api/hotel-client';
+
+const validationSchema = yup.object({
+    name: yup
+        .string('Enter hotel name')
+        .required('Hotel name is required'),
+    address: yup
+        .string('Enter address'),
+    contactName: yup
+        .string('Enter contact name'),
+    email: yup
+        .string('Enter email')
+        .email('Enter a valid email'),
+    phone: yup
+        .string('Enter phone'),
+    website: yup
+        .string('Enter website'),
+});
+
+const SaveHotelForm = ({ isOpen, handleSaveHotelModalToggle, hotelModel }) => {
+    const isEditForm = Boolean(hotelModel && hotelModel.id);
+    const initialValues = hotelModel || {
+        name: '',
+        stars: 0,
+        address: '',
+        contactName: '',
+        email: '',
+        phone: '',
+        website: '',
+        attachments: [],
+    };
+    const formik = useFormik({
+        initialValues: initialValues,
+        validationSchema: validationSchema,
+        onSubmit: async (model, formikHelper) => {
+            const newHotel = isEditForm
+                ? await HotelClient.editHotel(model)
+                : await HotelClient.addHotel(model);
+
+            if (newHotel) {
+                formikHelper.resetForm();
+            }
+        },
+        onReset: () => {
+            handleSaveHotelModalToggle();
+        }
+    });
+
+    function handleFileAdd(file) {
+        formik.setFieldValue('attachments', [...formik.values.attachments, { name: file.name, blob: file }]);
+    }
+
+    function handleAttachmentRemove(attachment) {
+        const attachments = formik.values.attachments;
+        const index = formik.values.attachments.indexOf(attachment);
+        if (index > -1) {
+            attachments.splice(index, 1);
+            formik.setFieldValue('attachments', attachments);
+        }
+    }
+
+    const inputMb = 1;
+    const dialogTitle = isEditForm ? 'Edit new hotel' : 'Add new hotel';
+
+    return (
+        <Dialog open={isOpen} maxWidth="lg">
+            <DialogTitle><HotelIcon />{dialogTitle}</DialogTitle>
+            <DialogContent>
+                <Container>
+                    <form onSubmit={formik.handleSubmit} noValidate autoComplete="off">
+                        <Grid container direction="row" justify="center" alignItems="flex-start" spacing={3}>
```

Comment:
`direction="row" justify="center" alignItems="flex-start"` are redundant for this case.